File: Java/DiscountStrategy.java
// DiscountStrategy.java
import java.math.BigDecimal;

/**
 * 割引計算の戦略を定義するインターフェース
 * ラムダ式として実装されることを想定
 */
@FunctionalInterface
public interface DiscountStrategy {
    BigDecimal applyDiscount(BigDecimal originalPrice);
}

File: PHP/DiscountStrategy.php
<?php
// DiscountStrategy.php

interface DiscountStrategy
{
    /**
     * 割引を適用した後の価格を計算する
     * @param float $originalPrice 元の価格
     * @return float 割引適用後の価格
     */
    public function apply(float $originalPrice): float;
}
